test/generator/example_generator.cpp

```diff
@@ -81,13 +81,13 @@ class Example : public Halide::Generator<Example> {
             .bound(c, 0, channels)
             .reorder(c, x, y)
             .unroll(c);
-        if (vectorize) {
-            // Note that we can use the Generator method natural_vector_size()
-            // here; this produces the width of the SIMD vector being targeted
-            // divided by the width of the data type.
-            output
-                .vectorize(x, natural_vector_size(output.type()));
-        }
+        // Note that we can use the Generator method natural_vector_size()
+        // here; this produces the width of the SIMD vector being targeted
+        // divided by the width of the data type.
+        // Use '1' as the 'no vectorize' case.
+        // TODO: this is awkward, there should be a more idiomatic way.
+        output
+            .vectorize(x, select(vectorize, natural_vector_size(output.type()), 1));
```

Comment:
Another thought: this isn't adding a new combinatoric problem, it's just moving it from what was GeneratorParams before, to specializations with this new feature.